In [9]:
import numpy as np
from scipy import stats

#### 1. Провести дисперсионный анализ для определения того, есть ли различия среднего роста среди взрослых футболистов, хоккеистов и штангистов. Даны значения роста в трех группах случайно выбранных спортсменов: Футболисты: 173, 175, 180, 178, 177, 185, 183, 182. Хоккеисты: 177, 179, 180, 188, 177, 172, 171, 184, 180. Штангисты: 172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170.

In [31]:
ft = np.array([173, 175, 180, 178, 177, 185, 183, 182])
hk = np.array([177, 179, 180, 188, 177, 172, 171, 184, 180])
st = np.array([172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170])
k = 3
n = len(np.hstack([ft, hk, st]))

In [11]:
stats.f_oneway(ft, hk, st)

F_onewayResult(statistic=5.500053450812596, pvalue=0.010482206918698694)

In [33]:
k-1, n-k

(2, 25)

1. Если сравнивать с таблицей крит. точек распределения Ф-С для уровня значимости 0.05, Fтабл = 3.4, что меньше Fрасч = 5.5.
2. Просто p-value = 0.01, что меньше уровня значимости 0.05

Принимаем гипотезу H1 = есть различия в росте у этих 3 групп спортсменов на уровне значимости 0.05